contracts/libs/Witnet.sol

```diff
@@ -3,314 +3,370 @@
-  /// @return The `string[]` decoded from the WitnetData.Result.
-  function asStringArray(WitnetData.Result memory _result) external pure returns(string[] memory) {
-    require(_result.success, "Tried to read `string[]` value from errored WitnetData.Result");
-    return _result.value.decodeStringArray();
-  }
-
-  /// @notice Decode a natural numeric value from a WitnetData.Result as a `uint64` value.
-  /// @param _result An instance of WitnetData.Result.
-  /// @return The `uint64` decoded from the WitnetData.Result.
-  function asUint64(WitnetData.Result memory _result) external pure returns(uint64) {
-    require(_result.success, "Tried to read `uint64` value from errored WitnetData.Result");
-    return _result.value.decodeUint64();
-  }
+    /// Data kept in EVM-storage for every Request posted to the Witnet Request Board.
+    struct Request {
+        IWitnetRequest addr;    // The contract containing the Data Request which execution has been requested.
+        address requestor;      // Address from which the request was posted.
+        bytes32 codehash;       // Codehash of the Data Request which execution has been requested.
+        uint256 gasprice;       // Minimum gas price the DR resolver should pay on the solving tx.
+        uint256 reward;         // escrow reward to by paid to the DR resolver.
+    }
 
-  /// @notice Decode an array of natural numeric values from a WitnetData.Result as a `uint64[]` value.
-  /// @param _result An instance of WitnetData.Result.
-  /// @return The `uint64[]` decoded from the WitnetData.Result.
-  function asUint64Array(WitnetData.Result memory _result) external pure returns(uint64[] memory) {
-    require(_result.success, "Tried to read `uint64[]` value from errored WitnetData.Result");
-    return _result.value.decodeUint64Array();
-  }
+    /// Data kept in EVM-storage containing Witnet-provided response metadata and result.
+    struct Response {
+        address reporter;       // Address from which the result was reported.
+        bytes32 proof;          // Witnet-provided validation proof of the reported result.
+        uint256 epoch;          // Witnet epoch in which the reported result was actually finalized.        
+        bytes   result;         // Witnet-provided result value, encoded as bytes array.
+    }
 
-  /// @notice Convert a stage index number into the name of the matching Witnet request stage.
-  /// @param _stageIndex A `uint64` identifying the index of one of the Witnet request stages.
-  /// @return The name of the matching stage.
-  function stageName(uint64 _stageIndex) public pure returns(string memory) {
-    if (_stageIndex == 0) {
-      return "retrieval";
-    } else if (_stageIndex == 1) {
-      return "aggregation";
-    } else if (_stageIndex == 2) {
-      return "tally";
-    } else {
-      return "unknown";
+    /// Data struct containing actual result data provided by the Witnet DON.
+    struct Result {
```

Comment:
I'm not sure if it is necessary to change it, but it looks a bit confusing since only some lines above the result of the response is defined

Reply:
Indeed, `Response.result` is declared as `bytes` and `Result` contains the struct returned after passing these `bytes` to `WitnetParserLib.resultFromCborBytes()`. The reason for not declaring `Response.result` as `Witnet.Result` is because the WRB is/was not intended to perform any kind of parsing or decodification. 

I will add a more explicit comment to `struct Result`: "Data struct produced by the 'WitnetParserLib' when parsing a Witnet-provided raw result, encoded as bytes array."

Reply:
Then again, if couple `WitnetParserLib` to the WRB, and make it expose the whole `IWitnetRequestParser` functionality (public methods of WitnetParserLib), we could totally decouple `UsingWitnet` (ergo PriceFeed contracts) from the `WitnetParserLib`, making developers life a little bit easier, and only one entry-point address to know about for each supported network.
Should the IWitnetRequestParser implementation be upgraded or patched, the upgrade would be done through out the Proxy, without forcing developers to deploy new instances of their contracts because of this.